src/circuit/auto-relay.js

```diff
@@ -121,15 +124,21 @@ class AutoRelay {
     }
 
     // Create relay listen addr
-    let listenAddr, remoteMultiaddr
+    let listenAddr, remoteMultiaddr, remoteAddrs
 
     try {
-      const remoteAddrs = this._peerStore.addressBook.get(connection.remotePeer)
+      remoteAddrs = this._peerStore.addressBook.get(connection.remotePeer)
       // TODO: HOP Relays should avoid advertising private addresses!
       remoteMultiaddr = remoteAddrs.find(a => a.isCertified).multiaddr // Get first announced address certified
     } catch (_) {
       log.error(`${id} does not have announced certified multiaddrs`)
-      return
+
+      // Attempt first if existing
+      if (!remoteAddrs || !remoteAddrs.length) {
+        return
+      }
+
+      remoteMultiaddr = remoteAddrs[0].multiaddr
```

Comment:
I added a sorter. In follow up PRs, I will add the support for custom sort function for dial and custom filter for announce. Then, it will be bubbled up to here. But I wanted to keep things simple in this PR